cpp/src/column/column_factories.cpp

```diff
@@ -51,6 +51,8 @@ std::size_t size_of(data_type element_type)
 // Empty column of specified type
 std::unique_ptr<column> make_empty_column(data_type type)
 {
+  CUDF_EXPECTS(type.id() == type_id::EMPTY || !cudf::is_nested(type),
+               "Make empty column is invalid to call on nested types");
```

Comment:
I don't think this exception message is correct anyway, since there are two ways to fail the expectation: nested types and type.id() != type_id::EMPTY.

Reply:
It's correct - you can call with EMPTY.  The reason it has that extra check in there is that is_nested() calls the type_dispatcher, which doesn't let you pass EMPTY.